src/features/TokenMigration/components/TokenMigrationForm.tsx

```diff
@@ -0,0 +1,236 @@
+import isEmpty from 'lodash/isEmpty';
+import * as Yup from 'yup';
+import { bigNumberify } from 'ethers/utils';
+
+import translate, { translateRaw } from '@translations';
+import { SPACING } from '@theme';
+import { IAccount, Network, StoreAccount, Asset } from '@types';
+import { AccountDropdown, InlineMessage, AmountInput, Button, Tooltip } from '@components';
+import { validateAmountField } from '@features/SendAssets/components/validators/validators';
+import { isEthereumAccount } from '@services/Store/Account/helpers';
+import { StoreContext, AssetContext, NetworkContext } from '@services/Store';
+import { fetchGasPriceEstimates } from '@services/ApiService';
+import { getNonce } from '@services/EthService';
+import { ETHUUID, noOp, weiToFloat } from '@utils';
+import { getAccountsWithAssetBalance } from '@features/SwapAssets/helpers';
+
+import { ISimpleTxFormFull } from '../types';
+import { tokenMigrationConfig } from '../config';
+
+interface Props extends ISimpleTxFormFull {
+  isSubmitting: boolean;
+  onComplete(fields: any): void;
+  handleUserInputFormSubmit(fields: any): void;
+}
+
+interface UIProps {
+  network: Network;
+  relevantAccounts: StoreAccount[];
+  isSubmitting: boolean;
+  onComplete(fields: any): void;
+}
+
+const FormFieldItem = styled.fieldset`
+  margin-bottom: ${SPACING.LG};
+`;
+
+const FormFieldLabel = styled.label`
+  display: flex;
+  font-size: 1rem;
+  margin-bottom: ${SPACING.SM};
+  font-weight: 400;
+  align-items: center;
+  flex-wrap: wrap;
+`;
+
+const FormFieldSubmitButton = styled(Button)`
+  width: 100%;
+  display: inline-block;
+  &:disabled {
+    background-color: rgba(0, 122, 153, 0.3);
+  }
+`;
+
+const TokenMigrationForm = ({ isSubmitting, onComplete }: Props) => {
+  const { accounts } = useContext(StoreContext);
+  const { networks } = useContext(NetworkContext);
+  const network = networks.find((n) => n.baseAsset === ETHUUID) as Network;
+  const relevantAccounts = accounts.filter(isEthereumAccount);
+
+  return (
+    <TokenMigrationFormUI
+      isSubmitting={isSubmitting}
+      network={network}
+      relevantAccounts={relevantAccounts}
+      onComplete={onComplete}
+    />
+  );
+};
+
+export const TokenMigrationFormUI = ({
+  isSubmitting,
+  network,
+  relevantAccounts,
+  onComplete
+}: UIProps) => {
+  const { getAssetByUUID } = useContext(AssetContext);
+  const defaultAsset = (getAssetByUUID(tokenMigrationConfig.fromAssetUuid) || {}) as Asset;
+  const initialFormikValues: ISimpleTxFormFull = {
+    account: {} as StoreAccount,
+    amount: '0',
+    asset: defaultAsset,
+    nonce: '0',
+    gasPrice: '20',
+    address: '',
+    gasLimit: '',
+    network
+  };
+
+  const TokenMigrationFormSchema = Yup.object().shape({
+    amount: Yup.number()
+      .min(0, translateRaw('ERROR_0'))
+      .required(translateRaw('REQUIRED'))
+      .typeError(translateRaw('ERROR_0'))
+  });
+
+  return (
+    <div>
+      <Formik
```

Comment:
Can we use `useFormik` instead of having the nested renders in `Field` and `Form`?

Reply:
This will need to be done in PurchaseMembership flow as well